fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2908,6 +2908,20 @@ public Range getUnbuiltRange() {
         }
     }
 
+    /**
+     * Marks an index as readable, but if uniqueness violations is found - mark the index
+     * as unique pending instead of throwing a {@link RecordIndexUniquenessViolation} exception.
+     *
+     * @param index the index to mark readable
+     * @return a future that will either complete exceptionally if the index can not
+     * be made readable/unique-pending or will contain <code>true</code> if the store was modified
+     * and <code>false</code> otherwise
+     */
+    @Nonnull
+    public CompletableFuture<Boolean> markIndexReadableOrUniquePending(@Nonnull Index index) {
```

Comment:
Would there be reason to mark an index as `READABLE_UNIQUE_PENDING` even if it could be `READABLE` (or is already)? I think to get their now, you would have to add a uniqueness violation.
(Note: I already feel like `FDBRecordStore` has too many methods to make too many things just a method call, so I'm not really pushing for more)

Reply:
I can't think about a reason to degrade a readable index that doesn't involve disabling the index altogether (or removing the UNIQUE flag - is it possible to do it today without re-indexing?). 
FMVHO, unique indexes in either `WRITE_ONLY` or `READABLE_UNIQUE_PENDING` states should forbid adding a conflicting records (when detected - if write only). But that's should probably be a separate discussion..